run_riscv_tests.py

```diff
@@ -0,0 +1,174 @@
+        sh_strtab = bytes[sh_strtab_offset:sh_strtab_offset+sh_strtab_size]
+
+        # Get the symbol table
+        SHT_SYMTAB = 0x2
+        SHT_STRTAB = 0x3
+        symtab = None
+        strtab = None
+        for i in range(e_shnum):
+            section = shdr[i*SECTION_SIZE:(i+1)*SECTION_SIZE]
+            sh_type = int.from_bytes(section[0x04:0x08], endianness)
+            sh_offset = int.from_bytes(section[0x10:0x14], endianness)
+            sh_size = int.from_bytes(section[0x14:0x18], endianness)
+            if sh_type == SHT_SYMTAB:
+                symtab = bytes[sh_offset:][:sh_size]
+            if sh_type == SHT_STRTAB:
+                strtab = bytes[sh_offset:][:sh_size]
+            if symtab != None and strtab != None:
+                break
+
+        # Walk the symbol table looking for tohost
+        SYM_ENTR_SIZE = 0x10
+        for i in range(1, len(symtab) // SYM_ENTR_SIZE):
+            sym_tab_entry = symtab[i*SYM_ENTR_SIZE:(i+1)*SYM_ENTR_SIZE]
+            st_name = int.from_bytes(sym_tab_entry[0x00:0x04], endianness)
+            st_value = int.from_bytes(sym_tab_entry[0x04:0x08], endianness)
+            if str(strtab[st_name:][:len("tohost")], 'utf-8') == "tohost":
+                tohost_address = st_value
+    return tohost_address
+
+def run_test(fname):
+    tohost_address = get_hostaddress(fname)
+    res = subprocess.run([verilator_binary, '--tohost-address', str(tohost_address), fname], stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+    if res.returncode != 0:
+        print('Verilator failed to run, exiting: ')
+        print(res.stderr)
+        exit(1)
+
+    if 'PASSED' not in res.stdout:
+        print(res.stdout)
+
+    return ('PASSED' in res.stdout)
+
+def run_selected_tests(isa, envs, machine_mode_tests):
+    pass_count = 0
+    total_count = 0
+
+    for env in envs:
+        print(f"Running '{env}' TVM tests...")
+        for ext in isa:
+            print(f"ISA: RV32{ext.capitalize()}")
+
+            tests = test_base_dir.glob(f"rv32u{ext}-{env}-*.bin")
+            for test in filter(apply_skips, tests):
+                total_count += 1
+                status = run_test(test)
+                if status:
+                    pass_count += 1
+                else:
+                    print(f"{Colors.RED}[FAILED]: {Colors.END}{test}")
+
+        # machine-mode tests
+        if machine_mode_tests:
+            print(f"Machine Mode tests")
+            tests = test_base_dir.glob(f'rv32mi-{env}*.bin')
+            for test in filter(apply_skips, tests):
+                total_count += 1
+                status = run_test(test)
+                if status:
+                    pass_count += 1
+                else:
+                    print(f"{Colors.RED}[FAILED]: {Colors.END}{test}")
+        
+    if pass_count == total_count:
+        print(f"{Colors.GREEN}[All tests passed.]{Colors.END}")
```

Comment:
`All {pass_count} tests passed` might be better because someone running this might think that everything got skipped and spend a bunch of time being confused (couldn't be me).